File: src/py/pg/prbm_greed/SafetyBoat.py
def solution(people, limit):
    firstIndex = 0
    lastIndex = len(people)-1
    count = 0    
    people.sort()
    
    while firstIndex <= lastIndex :
        if people[firstIndex]+people[lastIndex] <= limit :
            firstIndex += 1
            lastIndex -= 1 #
        else :
            lastIndex -= 1 # 이 부분 중복이라 한 블럭 바깥으로 빼내도 된다
        count += 1

    return count

File: src/jv/pg/prbm_greed/SafetyBoat.java
package jv.pg.prbm_greed;

import java.util.Arrays;

public class SafetyBoat {

public static void main(String[] args) {
	
}

public int solution(int[] people, int limit) {
	int cnt=0;
	Arrays.parallelSort(people);
	int peopleLeft = people.length;
	int startIndex = 0;
	int endIndex = people.length-1;
	while(peopleLeft>0) {
		if(people[endIndex]+people[startIndex]<=limit) {
    		--endIndex;
    		++startIndex;        		
    		peopleLeft-=2;
    	} else {
    		--endIndex;
    		peopleLeft-=1;
    	}
		++cnt;
	}
    return cnt;
}
}